CMakeLists.txt

```diff
@@ -195,72 +195,60 @@ target_link_libraries(
         numa
 )
 
-# Note on flags:
-#   ttnn and tt_lib will break if built with LTO, so leaving -fno-lto in compile options
-add_library(linker_flags INTERFACE)
-
-add_library(compiler_warnings INTERFACE)
-target_compile_options(
-    compiler_warnings
-    INTERFACE
-        -Werror
-        -Wdelete-non-virtual-dtor
-        -Wreturn-type
-        -Wswitch
-        -Wuninitialized
-        -Wno-unused-parameter
-)
-
-# add additional compile warning flags depending on the compiler
-ADJUST_METAL_COMPILER_WARNINGS()
-
-add_library(compiler_flags INTERFACE)
-target_link_libraries(
-    compiler_flags
-    INTERFACE
-        linker_flags
-        compiler_warnings
+string(TOUPPER "$ENV{ARCH_NAME}" ARCH_NAME_DEF)
+add_compile_definitions(ARCH_${ARCH_NAME_DEF})
+add_compile_options(
```

Comment:
This won't propagate on consumers, right?

e.g. I sure recommend everyone to use werror, but I am not bold enough to enforce it on all consumers of my code.

Reply:
Correct. This is (now) directory scope.  We process our dependencies before setting these, so those will be built with their own defaults.  Then we set these to become defaults for all our stuff.

This is a slight change in behaviour because previously we've been exporting these as part of our targets.  So consumers pulling in tt-metal as a submodule were getting these things shoved down their throats 😢 

We may need to publish certain flags.  But when those come up we should do them properly and not push non-mandatory flags.